y_py/websocket_provider.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+from typing import Any, Dict
+
+from websockets import connect
+from .y_py import *
+
+from .yutils import YMessageType, create_sync_step1_message, create_sync_step2_message, create_update_message, get_message
+
+
+class YwsDoc:
+    def __init__(self, ydoc, websocket):
+        self._ydoc = ydoc
+        self._websocket = websocket
+
+    def begin_transaction(self):
+        return Transaction(self._ydoc, self._websocket)
+
+    def get_map(self, name):
```

Comment:
I couldn't subclass YDoc because it's not a pure Python class I think. Ideally I would like to do:
```py
class YwsDoc(YDoc):
    ...
```
I just forwarded the `get_map` method for now, but would need to do it for every YDoc method.